context.go

```diff
@@ -65,13 +65,14 @@ type context struct {
 
 	msg      *message
 	done     bool
-	ackSent  bool
 	counters struct {
-		calls     int32
-		callsDone int32
+		calls     int
+		callsDone int
+		stores    int
 	}
-	m  sync.Mutex
-	wg *sync.WaitGroup
+	errors Errors
+	m      sync.Mutex
```

Comment:
This mutex can be substituted by a `RWMutex`, considering that you're need create a method like: 
```go
func (ctx Context) Stores() int {
	ctx.m.RLock()
	defer ctx.m.RUnlock()
	return ctx.counters.stores
}
```
The use of a `RWMutex` is recommended when for this case.